src/evaluations/data/evaluation_configs.py

```diff
@@ -528,6 +528,47 @@ def _generate_freq_configs_scenario_2(universe_size, num_sets, set_size):
   return scenario_config_list
 
 
+def _generate_freq_configs_scenario_3(universe_size, num_sets, set_size):
+  """Generate configs of Frequency Scenario 3.
+
+  This is a stress testing, in which each publisher serves a fixed number of
+  impressions to every reached id.
+
+  See Frequency Scenario 3: Per-publisher frequency capping:
+  https://github.com/world-federation-of-advertisers/cardinality_estimation_evaluation_framework/blob/master/doc/cardinality_and_frequency_estimation_evaluation_framework.md#frequency-scenario-3-per-publisher-frequency-capping
+
+  Args:
+    universe_size: the universal size of reach.
+    num_sets: the number of sets.
+    set_size: size of each set, assuming they're all equal.
+
+  Returns:
+    A list of ScenarioConfigs of freq scenario 3 per-publisher frequency
+    capping.
+  """
+  frequency_list = [2, 3, 5, 10]
+  scenario_config_list = []
+  for frequency in frequency_list:
+    scenario_config_list.append(
+        ScenarioConfig(
+            name='-'.join([
+                'per_publisher_frequency_cap',
```

Comment:
Maybe use consistent terms. Either frequency_cap or constant_frequency. I prefer constant_frequency because frequency_cap can mean upper or lower cap.

Reply:
Done.